agent/agent.go

```diff
@@ -1617,6 +1622,26 @@ LOAD:
 		}
 	}
 
+	// Only perform the following checks if there was an encrypt_key
+	// provided in the configuration.
+	if a.config.EncryptKey != "" {
+		msg := " keyring doesn't include key provided with -encrypt, using keyring"
+		if existingLANKeyring &&
+			keyringIsMissingKey(
+				config.SerfLANConfig.MemberlistConfig.Keyring,
+				a.config.EncryptKey,
+			) {
+			a.logger.Warn("LAN" + msg)
```

Comment:
nit:

I think this would be more consistent with existing logs if it were:

```
a.logger.Warn(msg, "keyring", "WAN")
```

During the hclog refactor it looked like most of the places that we had previously formatted a log message we turned into having a constant string message and passing the other data to the logger as kv pairs.

The same thing goes for the log for the "LAN" case too. If you really prefer this form, then thats fine with me too. My opinions on the subject are not very strong either way.

Reply:
Thank you Matt! This was actually a thing I kept wondering how to do with the new logger! I like your suggestion and will go with it!

Reply:
:thumbsup: